tests/run-pass/non_extern_foreign_call.rs

```diff
@@ -0,0 +1,39 @@
+#![feature(rustc_attrs)]
+
+#[no_mangle]
+extern "C" fn foo() -> i32 {
+    -1
+}
+
+#[export_name = "bar"]
+fn bar() -> i32 {
+    -2
+}
+
+#[rustc_std_internal_symbol]
```

Comment:
I am not sure if it is worth testing internal attributes like this.
OTOH, maybe this means we can remove some of the forwarding code like

https://github.com/rust-lang/miri/blob/a86eab3e6c3d645769bfdb2d09efd84d1e1fcbc0/src/shims/foreign_items.rs#L140

https://github.com/rust-lang/miri/blob/a86eab3e6c3d645769bfdb2d09efd84d1e1fcbc0/src/shims/foreign_items.rs#L172

Reply:
It seems I can't remove `panic_impl` (it complains about "can't call foreign function"). I removed the code for the two panic functions though.

Reply:
Ah, I think `panic_impl` has some additional magic directly on the LLVM level that wires it up with the lang item, so we need similar magic here.

> I removed the code for the two panic functions though.

Nice. :)